src/Adapter/Carrier/Repository/CarrierRepository.php

```diff
@@ -96,18 +96,40 @@ public function updateInNewVersion(CarrierId $carrierId, Carrier $carrier): Carr
         if ($carrier->position) {
             $newCarrier->position = $carrier->position;
         }
-        if ($carrier->active) {
+        if (null !== $carrier->active) {
             $newCarrier->active = $carrier->active;
         }
         if ($carrier->delay) {
             $newCarrier->delay = $carrier->delay;
         }
+
+        if (null !== $carrier->shipping_handling) {
+            $newCarrier->shipping_handling = $carrier->shipping_handling;
+        }
+        if (null !== $carrier->is_free) {
+            $newCarrier->is_free = $carrier->is_free;
+        }
+        if (null !== $carrier->shipping_method) {
+            $newCarrier->shipping_method = $carrier->shipping_method;
+        }
+        if (null !== $carrier->range_behavior) {
+            $newCarrier->range_behavior = $carrier->range_behavior;
+        }
+
         $newCarrier->deleted = false; // just to be sure...
 
         // Copy all others information like ranges, shops associated, ...
         $newCarrier->copyCarrierData($carrierId->getValue());
+
         $this->updateObjectModel($newCarrier, CannotUpdateCarrierException::class);
 
         return new CarrierId($newCarrier->id);
     }
+
+    public function setTaxRuleGroup(CarrierId $carrierId, int $idTaxRuleGroup): void
```

Comment:
I don't think this is efficient to fetch an object everytime the tax rules need to be updated, because you're supposed to have the object instance already and you'd just need to update it

This is similar to @tleon 's PR https://github.com/PrestaShop/PrestaShop/pull/36246 that is still in progress, but the idea is that `updateInNewVersion` should return the new instance so you can use it afterwards without re-fetching it systematically

Reply:
This would be the last big thing to handle, since Thomas's PR is still pending you should probably copy the part from his PR so that we can merge this one sooner

Reply:
Still needs to be done, either by copying the code from the pending PR or then this one is blocked until the other is merged